R/create_base_vectors.R

```diff
@@ -6,294 +6,416 @@
+    if (!is.numeric(epsg)) {
+      stop("There was a problem retrieving the EPSG code from the aoi. Is it assigned properly?")
+    }
+  }
+
+  epsg <- sf::st_crs(aoi)
+
+  # note this was an earlier flag - I dont like the idea of changing internals
+  # @Andy T. any thoughts on speed improvements.
+
+  # Adjust max download size based on AOI
+  ## PROBLEMATIC -- should not be done globally -----------------
+  ## see: https://r-pkgs.org/code.html#sec-code-r-landscape section 7.6.1
+  # withr::local_options(
+  #   options(bcdata.max_geom_pred_size = as.numeric(st_area(aoi)) + 10)
+  # )
+
+  get_BEC(aoi, out_dir)
+  get_VRI(aoi, out_dir)
+  get_harvest(aoi, out_dir)
+  get_TEM(aoi, out_dir)
+  get_water(aoi, out_dir)
+  get_roads(aoi, out_dir)
+  get_towns(aoi, out_dir)
+  get_fires(aoi, out_dir)
+  get_fire_severity(aoi, out_dir)
+  get_parks(aoi, out_dir)
+  get_transmission_lines(aoi, out_dir)
+
+  cli::cat_line()
+  cli::cli_alert_success(
+    "Layers downloaded and to written to {.path {out_dir}}"
+  )
+}
+
+### 1) Get_BEC ----------------------------
+get_BEC <- function(aoi, out_dir) {
+  #message("\rDownloading BEC layers")
+
+  # # 1) BEC Biogeographical linework
+  bec <- bcdata::bcdc_query_geodata("f358a53b-ffde-4830-a325-a5a03ff672c3") |>
+    bcdata::filter(bcdata::INTERSECTS(aoi)) |>
+    bcdata::select("MAP_LABEL") |>
+    bcdata::collect() |>
+    dplyr::select("MAP_LABEL") %>%
+    {
+      if (nrow(.) > 0) sf::st_intersection(., aoi) else .
+    }
+
+  if (sf::st_crs(aoi) == sf::st_crs(bec)) {
+    sf::st_write(bec, fs::path(out_dir, "bec.gpkg"), append = FALSE)
+    cli::cat_line()
+    cli::cli_alert_success(
+      "Bec layer downloaded and to written to {.path {out_dir}}"
+    )
+  } else {
+    # add conversion
+    message("convert or setting crs for bec data download")
+  }
+}
+
+
+### 2) Download VRI -----------------------
+get_VRI <- function(aoi, out_dir) {
+  message("\rDownloading VRI layers")
+
+  vri <- bcdata::bcdc_query_geodata("2ebb35d8-c82f-4a17-9c96-612ac3532d55") |>
+    bcdata::filter(bcdata::INTERSECTS(aoi)) |>
+    bcdata::select(c("BCLCS_LEVEL_2", "BCLCS_LEVEL_4", "PROJ_AGE_CLASS_CD_1", "SPECIES_CD_1")) |> # Treed sites
+    bcdata::collect() |>
+    dplyr::select(c("BCLCS_LEVEL_2", "BCLCS_LEVEL_4", "PROJ_AGE_CLASS_CD_1", "SPECIES_CD_1")) %>%
+    {
+      if (nrow(.) > 0) sf::st_intersection(., aoi) else .
+    }
+
+  sf::st_write(vri, fs::path(out_dir, "vri.gpkg"), append = FALSE)
+  cli::cat_line()
+  cli::cli_alert_success(
+    "VRI layer downloaded and to written to {.path {out_dir}}"
+  )
+  # post process VRI data into classes:
+
+  # Depending on the study area we want to focus on sampling in older areas - class 4 (60-80) or 5 (80 + )
+  # To do this we will define 2 vri age classess to exclude from the sampling area
+  # class 1 and 2 (0 - 40 yrs)
+  # class 1-3 (0 - 60 years)
+
+  vri_class2 <- vri |>
+    dplyr::mutate(age_class = as.numeric(PROJ_AGE_CLASS_CD_1)) |>
```

Comment:
When you run `check()` you get lots of "no visiible binding for global variable". Unquoted variable names like these should be used with the `.data$` pronoun:

```suggestion
    dplyr::mutate(age_class = as.numeric(.data$PROJ_AGE_CLASS_CD_1)) |>
```

And in some package-level documentation (e.g., if you run `usethis::use_package_doc()`, you need to add:

```r
#' @importFrom rlang .data
```

Reply:
See https://dplyr.tidyverse.org/articles/in-packages.html#data-masking-and-tidy-selection-notes